lib/Sema/TypeOfReference.cpp

```diff
@@ -2191,11 +2191,28 @@ void ConstraintSystem::bindOverloadType(const SelectedOverload &overload,
                            // FIXME: Should propagate name-as-written through.
                            : DeclNameRef(choice.getName());
 
-    addValueMemberConstraint(
-        LValueType::get(rootTy), memberName, memberTy, useDC,
-        isSubscriptRef ? FunctionRefInfo::doubleBaseNameApply()
-                       : FunctionRefInfo::unappliedBaseName(),
-        /*outerAlternatives=*/{}, keyPathLoc);
+    // Check the current depth of applied dynamic member lookups, if we've
+    // exceeded the limit then record a fix and set a hole for the member.
+    unsigned lookupDepth = [&]() {
+      auto path = keyPathLoc->getPath();
+      auto rbegin = path.rbegin();
+      auto rend = path.rend();
+      auto iter = std::find_if(rbegin, rend, [&](const LocatorPathElt &elt) {
```

Comment:
I think we can use `keyPathLoc->findLast<LocatorPathElt::KeyPathDynamicMember>(<<iter>>)` in here?

Reply:
Hmm it's not quite the same thing since we want the last element that _isn't_ a dynamic member, although maybe it's worth adding `findFirst/findLast` overloads for that? Could also use in `isSubscriptMemberRef`